app/src/main/java/org/jellyfin/androidtv/ui/itemhandling/BaseRowItem.java

```diff
@@ -493,13 +495,20 @@ public Drawable getBadgeImage(Context context) {
     public void refresh(final EmptyResponse outerResponse) {
         switch (type) {
             case BaseItem:
-                apiClient.getValue().GetItemAsync(getItemId(), KoinJavaComponent.<UserRepository>get(UserRepository.class).getCurrentUser().getValue().getId().toString(), new Response<BaseItemDto>() {
-                    @Override
-                    public void onResponse(BaseItemDto response) {
-                        baseItem = response;
-                        outerResponse.onResponse();
-                    }
-                });
+                try {
+                    apiClient.getValue().GetItemAsync(getItemId(), KoinJavaComponent.<UserRepository>get(UserRepository.class).getCurrentUser().getValue().getId().toString(), new Response<BaseItemDto>() {
```

Comment:
Since we only want to catch exceptions for the `KoinJavaComponent.<UserRepository>get(UserRepository.class).getCurrentUser().getValue().getId().toString()` call it might be better to assign the result to a variable first and only try-catch that (with a return/break in the catch)

Another, maybe even better, solution is to null check the getValue() call.

Reply:
GetItemAsync alks throws IllegalArgumentException, which i also wanted to catch.
Can add the specific exceptions.

Reply:
Do you have a stacktrace of the IllegalArgumentException? The linked issues only have NPE's. And yes catching specific exceptions is preferred.

Reply:
Catching invalid id's just seems like a reasonable precaution, until we fixed the root cause.

So i was thinking we should not trust any parameter for this async call and catch everything. That's why i specifically print the exception's for debug, so we can still try to fix the issue eventually.